### PR TITLE
Add typescript definition file

### PR DESCRIPTION
**Pull request by jnpdx**
# Summary

This adds an `index.d.ts` Typescript definition file to the project.   The typings are taken either directly from Typescript's inferences of the functions or from React Native's types via `@types/react-native`

This adds out-of-the-box Typescript support.  Note that it does depend on @types/react-native, although I did not add this to `peerDependencies` since only Typescript users need this and would obviously have it installed already.

### What's required for testing (prerequisites)?

Add `react-native-hooks` to a Typescript enabled React Native project.  

### What are the steps to reproduce (after prerequisites)?

Typings should be available with all of the same syntax/usage as before.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by paldepind** [comment]
@jnpdx That's even better :smile:

**Comment by marnusw** [comment]
Since `useDeviceOrientation` is a concept of this library's own creation and doesn't map directly to RN could we add an exported interface for its return value?

```ts
export interface DeviceOrientation {
  portrait?: boolean
  landscape?: boolean
}

declare function useDeviceOrientation(): DeviceOrientation
```

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  AppStateStatus,
+  ScaledSize,
+  KeyboardStatic,
+  GetPhotosReturnType,
+  LayoutRectangle
+} from "react-native";
+
+declare function useDimensions(): { window: ScaledSize; screen: ScaledSize };
+declare function useAppState(): AppStateStatus;
+declare function useBackHandler(handler: () => boolean);
```
Comment (marnusw) [inline index.d.ts]: Missing return type `: void`

**Comment by jnpdx** [comment]
New suggestions from @marnusw implemented

**Comment by asurare** [comment]
Why is this PR not merged ?

**Comment by pvinis** [comment]
I'm guessing people are waiting for a move to typescript? We should merge this anyway until this happens. Also I started a PR for a move to typescript and I would appreciate people's help there so we can finish quicker.

https://github.com/react-native-community/react-native-hooks/pull/41

**Comment by pvinis** [comment]
Closing this. Now that we moved to TS, this is generated. Thanks everyone for the input and help!

**Pull request closed without merging**